dhall/src/Dhall.hs

```diff
@@ -791,6 +798,77 @@ list = fmap Data.Foldable.toList . sequence
 vector :: Type a -> Type (Vector a)
 vector = fmap Data.Vector.fromList . list
 
+{-| Decode a `Set` from a `List`
+
+>>> input (set natural) "[1, 2, 3]"
+fromList [1,2,3]
+
+Duplicate elements are ignored.
+-}
+set :: (Ord a) => Type a -> Type (Data.Set.Set a)
+set = fmap Data.Set.fromList . list
```

Comment:
I think this should be the same as `setFromDistinctList`.  My reasoning is that by default we should fail loudly if we detect anything wrong

Reply:
Yeah, that seems reasonable to me